components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -139,8 +141,19 @@ private PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReque
             if (log.isDebugEnabled()) {
                 log.debug("Mandatory claims missing for the application : " + missingClaims[0]);
             }
-
             try {
+                // If there are read only claims marked as mandatory and they are missing, we cannot proceed further.
+                // We have to end the flow and show an error message to user.
+                ClaimManager claimManager = getUserRealm(context.getTenantDomain()).getClaimManager();
+                for (String claim : missingClaims[0].split(",")) {
```

Comment:
Avoid using missingClaims which use concatenated claims string which contains multiple claims..

Rather we need to iterate claims (from context may be) where it available claims as distincts items, rather splitting a concatenated string.